src/io/flutter/jxbrowser/JxBrowserManager.java

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2020 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.jxbrowser;
+
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.progress.ProgressIndicator;
+import com.intellij.openapi.progress.ProgressManager;
+import com.intellij.openapi.progress.Task;
+import com.intellij.openapi.progress.impl.BackgroundableProcessIndicator;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Pair;
+import com.intellij.util.containers.ContainerUtil;
+import com.intellij.util.download.DownloadableFileDescription;
+import com.intellij.util.download.DownloadableFileService;
+import com.intellij.util.download.FileDownloader;
+import com.intellij.util.lang.UrlClassLoader;
+import io.flutter.utils.FileUtils;
+import io.flutter.utils.JxBrowserUtils;
+import org.jetbrains.annotations.NotNull;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.atomic.AtomicReference;
+
+enum JxBrowserStatus {
+  NOT_INSTALLED,
+  INSTALLATION_IN_PROGRESS,
+  INSTALLED,
+  INSTALLATION_FAILED,
+}
+
+// JxBrowser provides Chromium to display web pages within IntelliJ. This class manages downloading the required files and adding them to
+// the class path.
+public class JxBrowserManager {
+  private static JxBrowserManager manager;
+  private static final String DOWNLOAD_PATH = FileUtils.platformPath();
+  private static final AtomicReference<JxBrowserStatus> status = new AtomicReference<>(JxBrowserStatus.NOT_INSTALLED);
+  private static final Logger LOG = Logger.getInstance(JxBrowserManager.class);
+  // We will be gating JxBrowser features until all of the features are landed.
+  // To test JxBrowser, set this to true and also add license key to VM options (-Djxbrowser.license.key=<key>).
+  public static final boolean ENABLE_JX_BROWSER = false;
+
+  private JxBrowserManager() {}
+
+  public static JxBrowserManager get() {
```

Comment:
Nit: for consistency with singletons in the rest of the code base call this getInstance() instead of get().